### DISCUSSION TITLE
Add explorer API URLs to `viem/chains`

### DISCUSSION THREAD

**Discussion by typedarray** [wevm/viem#1681] [Idea / Feature Request]
We (Ponder) currently maintain a list of explorer API URLs for a handful of popular chains:

```
  "etherscan.io": {
    name: "mainnet",
    chainId: 1,
    apiUrl: "https://api.etherscan.io/api",
  },
  "ropsten.etherscan.io": {
    name: "ropsten",
    chainId: 3,
    apiUrl: "https://api-ropsten.etherscan.io/api",
  },
  "rinkeby.etherscan.io": {
    name: "rinkeby",
    chainId: 4,
    apiUrl: "https://api-rinkeby.etherscan.io/api",
  }
```

Would you be open to a contribution upstreaming this to `viem/chains`? We could add an `apiUrl?: string;` field to the ChainBlockExplorer type:

https://github.com/wevm/viem/blob/c10c68eff0d02f0dcc0970bcceddd026801b27bd/src/types/chain.ts#L72-L75

Chain definitions would look like:

```ts
export const mainnet = defineChain({
  id: 1,
  // ...
  blockExplorers: {
    default: {
      name: 'Etherscan',
      url: 'https://etherscan.io',
      apiUrl: 'https://api.etherscan.io/api',
    },
  },
})

export const zora = defineChain({
  id: 7777777,
  // ...
  blockExplorers: {
    default: {
      name: 'Zora Explorer',
      url: 'https://explorer.zora.energy',
      apiUrl: 'https://explorer.zora.energy/api',
    },
  },
})
```

Taking it a step further, we could label which kind/brand each explorer is (etherscan, blockscout, etc).

**Comment by tmm** [comment]
> Would you be open to a contribution upstreaming this to viem/chains? We could add an apiUrl?: string; field to the ChainBlockExplorer type

This sounds okay.

> Taking it a step further, we could label which kind/brand each explorer is (etherscan, blockscout, etc).

Let's just keep single default option for now. We used to include multiple block explorers per chain (if they existed), but decided to trim things down in v2 since it's a lot of work to manage.

**Comment by typedarray** [comment]
Added in https://github.com/wevm/viem/pull/1687
